### PR TITLE
DPAT-546

### PR DESCRIPTION
**Pull request by SerbanCatrava**

### CODE REVIEW

**Comment by xoltar** [comment]
Not everyone has access to the DPAT issue tracker, would be helpful to copy the description from that issue here.

File: engine-plugins/frame-plugins/src/main/scala/org/trustedanalytics/atk/engine/frame/plugins/load/JdbcPlugin/JdbcConnectionFactory.scala [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+   * Create a connection url string for jdbc
+   * @param arguments required arguments
+   * @return a connection url
+   */
+  def connectionUrl(arguments: JdbcArgs): String = {
+
+    var url: String = "jdbc:"
+
+    //connectionUrl
+    if (isMySql(arguments.driverType)) {
```
Comment (xoltar) [inline engine-plugins/frame-plugins/src/main/scala/org/trustedanalytics/atk/engine/frame/plugins/load/JdbcPlugin/JdbcConnectionFactory.scala]: You could probably have a DriverType algebraic datatype (sealed trait DriverType with two case objects that extend it, one for mysql and one for sql server). That way the trait could define a connectionUrl method, and each drivertype could implement the method as appropriate. That way you could avoid all these if statements and just say: val url = arguments.driverType.connectionUrl(arguments). Plus you'd get the validation for only using supported driver types for free.

File: engine-plugins/frame-plugins/src/main/scala/org/trustedanalytics/atk/engine/frame/plugins/load/JdbcPlugin/LoadJdbcImpl.scala [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    sqlContext.load("jdbc", options)
+  }
+
+  /**
+   * Converts a jdbc data type to DataType
+   * @param dbType jdbc type
+   * @return a DataType
+   */
+  def dbToDataType(dbType: String): DataType = {
```
Comment (xoltar) [inline engine-plugins/frame-plugins/src/main/scala/org/trustedanalytics/atk/engine/frame/plugins/load/JdbcPlugin/LoadJdbcImpl.scala]: There should be a way to solve this without the chained ifs as well...

File: engine/engine-core/src/main/scala/org/apache/spark/frame/FrameRdd.scala [hunk 3]
```diff
@@ -496,6 +496,26 @@ object FrameRdd {
   }
 
   /**
+   * Converts a spark dataType (as string)to our schema Datatype
```
Comment (joyeshmishra) [inline engine/engine-core/src/main/scala/org/apache/spark/frame/FrameRdd.scala:499]: Let's not put Datatype conversion methods here. Use parseMany routine from DataTypes.scala. Same applies for LoadHBaseImpl.scala. We should fix both.

File: engine/interfaces/src/main/scala/org/trustedanalytics/atk/domain/DomainJsonProtocol.scala [hunk 4]
```diff
@@ -562,5 +562,6 @@ object DomainJsonProtocol extends AtkDefaultJsonProtocol with EventLogging {
 
   implicit val hBaseArgsSchemaFormat: RootJsonFormat[HBaseSchemaArgs] = jsonFormat3(HBaseSchemaArgs)
```
Comment (joyeshmishra) [inline engine/interfaces/src/main/scala/org/trustedanalytics/atk/domain/DomainJsonProtocol.scala:563]: Just for clarity. Could we remove the RootJsonFormat[<>]. It is implicit that we define jsonSchema and need not say the type explicitly. If you see the file, look at how the items above this are defined.

File: python-client/hbase.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# coding: utf-8
+import trustedanalytics as tp
```
Comment (joyeshmishra) [inline python-client/hbase.py:2]: *\* This file should not be there.**

**Comment by joyeshmishra** [comment]
Initial commit looks good. Don't merge now.
What won't work:
1. This will not work for any database other than Postgres. Install MySQL or any other database and try it out in Yarn Cluster mode. You will encounter the issue of shipping the libraries. Check how Hive plugin implements it
2. We should test the method with both jdbc and non jdbc urls.
3. Configuration changes for DP2 and non DP2 environments need to be made.
4. All databases do not have a standard jdbc URI. For e.g., refer: http://www.codejava.net/java-se/jdbc/jdbc-database-connection-url-for-common-databases . So building URI is going to be a challenge if it is not provided by user/environment.

**Pull request closed without merging**